src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,296 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager::AnalyzerManager(UserSettingsPointer pConfig) :
+    m_pConfig(pConfig),
+    m_nextWorkerId(0),
+    m_defaultTrackQueue(),
+    m_prioTrackQueue(),
+    m_defaultWorkers(),
+    m_priorityWorkers(),
+    m_pausedWorkers() {
+
+    int maxThreads = m_pConfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"));
+    int ideal = QThread::idealThreadCount();
+    if (ideal < 1) {
+        if (maxThreads > 0 && maxThreads <= 32) {
+            qDebug() << "Cannot detect idealThreadCount. maxThreads is: " << maxThreads;
+            ideal = maxThreads;
+        }
+        else {
+            qWarning() << "Cannot detect idealThreadCount. Using the sane value of 1";
+            ideal = 1;
+        }
+    }
+    if (maxThreads <= 0 || maxThreads > ideal) {
+        qWarning() << "maxThreads value is incorrect. Changing it to " << ideal;
```

Comment:
Oh, Yup, that would work too, although I meant where is the configuration initizalized for the first time, so that I could assign the ideal value to it at that time.